new/target_mode.md

```diff
@@ -0,0 +1,53 @@
+---
+RFC: unassigned
+Title: Chef Target Mode
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Chef Target Mode
+
+When deploying applications or systems, we frequently need to configure
+appliances where Chef cannot run natively. To allow us to do so, we need
+to run chef on a dedicated management node and interact with the
+appliance remotely, via an API.
+
+## Motivation
+
+    As a site reliability engineer,
+    I want to configure routers, switches and load balancers,
+    so that my application functions correctly.
+
+## Specification
+
+The concept of a node in chef is a 1:1 mapping with a client, enabling a
+single node to have several identities if needed. We can leverage this
+facility to create a target mode, allowing a management node to run chef
+for many appliances.
+Target mode is enabled by several new configuration options:
+ * `target`: The details for connectivity to the target
+   system. The result of this configuration option should be a
+   class instance that cookbooks can send messages to. This
+   enables target mode, disabling ohai entirely.
```

Comment:
I think a slightly better way to do this would be to define a lower-level transport which includes ways to mutate and inspect some remote target system which can be used for both ohai plugins and implementing resources/providers.

I'm thinking of something similar to [chef/train](https://github.com/chef/chef-rfc/pull/278/), which implements POSIX file operations, and allows you to run shell commands on the target system, and wholy defines how these calls get implemented on the target, be they simply to locally run the operations, or to forward them over ssh, or adb, or whatever.

I think that'd allow you to reuse a lot of existing providers with fairly minimal modifications, and I don't think it'd be too difficult to extend the Resource/Provider architecture to allow writing providers for specific target types, like Android phones over ADB, or a switch over ssh for instance.

Reply:
@zfjagann I think you might be reading a bit too far in to this. We are very much not planning for this to do transparent transport abstraction stuff. This is purely around running the client, what's actually in the recipe code is entirely up to those cookbooks.